### PR TITLE
refactor: Script for changing manifest versions for executors

### PR DESCRIPTION
**Pull request by rutujasurve94**

### CODE REVIEW

File: manifest-version.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+""" Script to change versioning of manifest.yml for executors
+[encoders, crafters, indexers, rankers]. Invoke this script from the
+executor directory passing old version and new version.
+Example usage from TextPaddleHubEncoder executor directory:
+python ../../../manifest-version.py
+"""
+import os, fnmatch
+old_version = "version: 0.0.1"
+latest_version = "version: 0.02"
```
Comment (JoanFM) [inline manifest-version.py:9]: it should be `0.0.2`?

Comment (rutujasurve94) [reply manifest-version.py]: changed to ```0.0.2``` for semantic versioning.

File: manifest-version.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+""" Script to change versioning of manifest.yml for executors
+[encoders, crafters, indexers, rankers]. Invoke this script from the
+executor directory passing old version and new version.
+Example usage from TextPaddleHubEncoder executor directory:
+python ../../../manifest-version.py
+"""
+import os, fnmatch
+old_version = "version: 0.0.1"
+latest_version = "version: 0.0.2"
+filePattern = "manifest.yml"
```
Comment (hanxiao) [inline manifest-version.py:10]: I would against using a hardcoded replacement for bumping version.

**Review by hanxiao — CHANGES_REQUESTED** [review]
the code should take care of: 
for each executor:
1. make the change by bumping the version 
2. create a branch, commit the code to that branch 
3. make a PR with that branch

File: manifest-version.py (hunk 2 above)
Comment (rutujasurve94) [reply manifest-version.py:15]: yes sure, removed all hardcoding

File: chore_version.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import sys
+repo_dir = os.getcwd()
+
+version_file = sys.argv[1]
+old_version = sys.argv[2]
+latest_version = sys.argv[3]
+repo = git.Repo.init(repo_dir)
+
+for dpath, dirs, files in os.walk(os.getcwd()):
+    for fname in fnmatch.filter(files, version_file):
```
Comment (hanxiao) [inline chore_version.py:20]: no need, simply use `glob.glob('**/manifest.yml')`

File: chore_version.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        fpath = os.path.join(dpath, fname)
+        dname = dpath.split("/")[-1]
+        print(fpath)
+		with open(fpath) as f:
```
Comment (hanxiao) [inline chore_version.py:24]: where is this indent come from?

File: chore_version.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		with open(fpath, "w") as f:
+		    f.write(s)
+		new_branch = repo.create_head('chore-version-' + dname)
+		new_branch.checkout()
+		index = repo.index
+		index.add(fpath)
+		index.commit("chore: bump version")
+		repo.git.push('--set-upstream', 'origin', new_branch)
+		title_string = "Upgrading manifest version for " + dname
+		body_string = "Changing version from 0.0.1 to 0.0.2"
```
Comment (hanxiao) [inline chore_version.py:36]: inconsistent to the old/new version that defined previously

File: chore_version.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		pr_command = "gh pr create --title \"" + title_string + "\" --body \"" + body_string + "\""
```
Comment (hanxiao) [inline chore_version.py:37]: this is very error-prone use string format

Comment (hanxiao) [reply chore_version.py:37]: `f'hello {my_var}'`

**Review by hanxiao — CHANGES_REQUESTED** [review]
see comments, also make sure you use single quote for string

**Comment by hanxiao** [comment]
for version bumping, can you do something like? 

```python
from ruamel.yaml import YAML

yaml = YAML()
with open('manifest.yml') as fp:
  info = yaml.load(fp)
  old_ver = info['version']
  new_ver = '.'join(str(int(old_ver.split('.')[-1])+1) + old_ver.split('.')[:-1])
```

File: chore_version.py (hunk 3 above)
Comment (rutujasurve94) [reply chore_version.py]: using glob.glob for finding file

File: chore_version.py (hunk 4 above)
Comment (rutujasurve94) [reply chore_version.py]: this doesn't show in local workspace but unusually showed on git

File: chore_version.py (hunk 5 above)
Comment (rutujasurve94) [reply chore_version.py]: Changed to old and new version instead of hard coding any version number

File: chore_version.py (hunk 6 above)
Comment (rutujasurve94) [reply chore_version.py]: using string format

**Comment by rutujasurve94** [comment]
Using yaml for version checks as per above

File: chore_version.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+version_file = sys.argv[1]
+repo = git.Repo.init(repo_dir)
+
+for fpath in glob.glob(f'**/{version_file}'):
+	dname = fpath.split("/")[-2]
+	with open(fpath) as fp:
+		info = yaml.load(fp)
+		old_ver = info['version']
+		new_ver = ''.join(str(int(old_ver.split('.')[-1])+1) + old_ver.split('.')[:-1])
```
Comment (hanxiao) [inline chore_version.py:26]:
```suggestion
		new_ver = '.'.join(old_ver.split('.')[:-1] + [str(int(old_ver.split('.')[-1])+1)])
```

File: chore_version.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	new_branch = repo.create_head('chore-version-' + dname)
```
Comment (hanxiao) [inline chore_version.py:27]:
```suggestion
	new_branch = repo.create_head(f'chore-version-{dname}')
```

File: chore_version.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+	new_branch.checkout()
+	index = repo.index
+	index.add(fpath)
+	index.commit("chore: bump version")
+	repo.git.push('--set-upstream', 'origin', new_branch)
+	title_string = f'Upgrading manifest version for {dname}'
+	body_string = "Bumping version"
+	pr_command = f'gh pr create --title \"{title_string}\" --body \"{body_string}\"'
```
Comment (hanxiao) [inline chore_version.py:35]:
```suggestion
	pr_command = f'gh pr create --title "{title_string}" --body "{body_string}"'
```

File: chore_version.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+		old_ver = info['version']
+		new_ver = ''.join(str(int(old_ver.split('.')[-1])+1) + old_ver.split('.')[:-1])
+	new_branch = repo.create_head('chore-version-' + dname)
+	new_branch.checkout()
+	index = repo.index
+	index.add(fpath)
+	index.commit("chore: bump version")
+	repo.git.push('--set-upstream', 'origin', new_branch)
+	title_string = f'Upgrading manifest version for {dname}'
+	body_string = "Bumping version"
```
Comment (hanxiao) [inline chore_version.py:34]:
```suggestion
	body_string = 'bumping version'
```

File: chore_version.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+		info = yaml.load(fp)
+		old_ver = info['version']
+		new_ver = ''.join(str(int(old_ver.split('.')[-1])+1) + old_ver.split('.')[:-1])
+	new_branch = repo.create_head('chore-version-' + dname)
+	new_branch.checkout()
+	index = repo.index
+	index.add(fpath)
+	index.commit("chore: bump version")
+	repo.git.push('--set-upstream', 'origin', new_branch)
+	title_string = f'Upgrading manifest version for {dname}'
```
Comment (hanxiao) [inline chore_version.py:33]:
```suggestion
	title_string = f'bumping version for {dname}'
```

File: chore_version.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	index = repo.index
+	index.add(fpath)
+	index.commit('chore: bump version')
+	repo.git.push('--set-upstream', 'origin', new_branch)
+	title_string = f'bumping version for {dname}'
+	body_string = 'bumping version'
+	pr_command = f'gh pr create --title "{title_string}" --body "{body_string}"'
+	# Command with shell expansion
+	subprocess.call(pr_command, shell=True)
+	repo.git.checkout('master')
```
Comment (hanxiao) [inline chore_version.py:41]: can you add a `break` here and run it for a test?

**Review by hanxiao — CHANGES_REQUESTED** [review]
can you add a `break` at the end of the for loop and run it for a test?

**Pull request merged**